lib/client.js

```diff
@@ -740,6 +740,7 @@ class Parser {
     if (!request) {
       return -1
     }
+    request.onResponseStarted()
```

Comment:
Can you add it to `http2` as well, there it should be within the following line: https://github.com/nodejs/undici/blob/107664ab220dc389c6dbb9c0f85a20cf0ba1ed3d/lib/client.js#L1784

As the `HEADERS` frame states the transition of the stream from `idle` to open, I believe it should be good to put it there.

Reply:
👍 I'll add it.